src/room_member.cpp

```diff
@@ -170,6 +171,7 @@ static void SendJoinRequest(RakNet::RakPeerInterface* peer, const std::string& n
 void RoomMember::ReceiveLoop() {
     // Receive packets while the connection is open
     while (IsConnected()) {
+        std::this_thread::sleep_for(sleep_time);
         std::lock_guard<std::mutex> lock(network_mutex);
```

Comment:
Mhh yep this is well defined it seems. I'd still prefer an explicit scope around the locked region so it's very obvious that we are creating a guard and when it will be destroyed.